### DISCUSSION TITLE
How to support Map type on isar 3.0.0 ?

### DISCUSSION THREAD

**Discussion by ductranit** [isar/isar#633] [Help] [answered]
Hi there,
Before I used `TypeConveter` to support map type (use json encode/decode).
But since the dev version 3.0.0-dev.9 `TypeConveter` was removed and replaced by `@embedded`
So I wonder how can I work with Map type by this way?
My properties look like:
```
Map<String, bool> checked = {};
Map<EnumType, dynamic> books = {};
```

**Comment by tannermeade** [comment, marked as answer]
Maybe "Ignore" the Map field, and use a getter/setter that converts the Map field into an Isar understandable type.

```
@Ignore()
Map<String, bool> checked = {};

String get isarChecked => jsonEncode(checked);
set isarChecked(String json) => checked = jsonDecode(json);
```

**Reply by simc** [reply]
Yeah that's exactly how you should do it 👍

**Reply by ductranit** [reply]
thank you, I'll try this
